pkg/sink/pulsar/factory.go

```diff
@@ -93,7 +95,10 @@ func setupAuthentication(config *config.PulsarConfig) (pulsar.Authentication, er
 	if config.AuthTLSCertificatePath != nil && config.AuthTLSPrivateKeyPath != nil {
 		return pulsar.NewAuthenticationTLS(*config.AuthTLSCertificatePath, *config.AuthTLSPrivateKeyPath), nil
 	}
-	return nil, fmt.Errorf("no authentication method found")
+
+	log.Warn("pulsar client no authentication method found")
+
+	return nil, nil
```

Comment:
should return error here ?

Reply:
Don't need an error, nil means no credentials are provided.

Reply:
No, pulsar supports to create a client without any Authentications